tests/json_test.go

```diff
@@ -112,6 +115,42 @@ func TestJSONOperations(t *testing.T) {
 			getCmd:   `JSON.GET escaped`,
 			expected: escapedCharsJSON,
 		},
+		{
```

Comment:
Add a new test function that validates error messages we are getting with the unsupported JSONPath patterns that you have mentioned in the PR description.
This would allow us to easily test if those patterns can be supported using some other library in the future.